File: easy/e9/python3/solution_test.py
import solution
import unittest


class TestSolution(unittest.TestCase):
    def test_twoSum(self):
        s = solution.Solution()

        input1 = [121, -121, 10]

        expectations = [True, False, False]

        if (len(input1) != len(expectations)):
            self.fail("Test is invalid")

        for i in range(len(expectations)):
            assert (-2**31 <= input1[i] and input1[i] <= 2**31)

            result = s.isPalindrome(input1[i])
            msg = f"bad result for {input1[i]}: got " +\
                f"{result} expected {expectations[i]}"

            self.assertEqual(result, expectations[i], msg)


if __name__ == '__main__':
    unittest.main()

File: easy/e9/typescript/solution_test.ts
import { isPalindrome } from "./solution"

function testIsPalindrome(): string {
    let input1: number[] = [123, 111, 101, 10, 11, -101, 12300321, 1, 0];

    let expectations: boolean[] = [false, true, true, false, true, false, true, true, true];

    if (input1.length != expectations.length) {
        return "test is invalid";
    }

    for (let i = 0; i < expectations.length; i++) {
        let result = isPalindrome(input1[i]);

        let msg = `bad result for ${input1[i]}: got ${result} expected ${expectations[i]}`;

        if (result != expectations[i]) {
            return msg;
        }
    }

    return "PASS";
}

console.log(testIsPalindrome());